### DISCUSSION TITLE
Automatic Searches - Infohash vs Path

### DISCUSSION THREAD

**Discussion by sotiris-bos** [cross-seed/cross-seed#806] [General]
Hello, I hadn't been using automatic searches until now and I have a question. 99% of my downloads are torrents.

My understanding is that Data-Based matching is preferable (but I don't know why).

However, the documentation recommends Infohash instead of Data (Path) for automatic searches.

Until now, I was using Data-based matching with symlinks to files. Now that I configured automatic searches with Infohash, the files are not getting symlinked and the new torrent simply points to the location of the original torrent.

What is preferred and why? Is the automatic search using Data (Path) the best way to configure it?

**Comment by zakkarry** [comment]
I'm not sure who told you it was preferred, or what your specific setup is that might benefit from one over the other. If it's just links that you want - then the v6 pre-release, which is near going to official stable release - will also support linking of torrent matches.

Data based matching is not "preferred" generally in v6, and we outline in the migration guide the main reason to use it going forward. Linking is not one of those reasons. We've always recommended other search methods (name or infohash) over data-based paths as the default, and that is why we recommend infohash, and not data based matching as the default go-to.

**Reply by sotiris-bos** [reply]
So, to my understanding now, data-based matches are preferred when the .torrent file is not available. If the .torrent file is available, then infohash is preferred?

And v6 will enable linking for infohash.

**Reply by zakkarry** [reply]
The linking behavior in v6 is covered in detail [here](https://www.cross-seed.org/docs/v6-migration#linking-updates)

The use cases for data based matching are covered [here](https://www.cross-seed.org/docs/v6-migration#usage-of-torrentdir-and-datadirs)

**Reply by zakkarry** [reply]
If you have a torrent loaded into your client, there is no reason you would need to use data based matching on that in v6.
